README.md

```diff
@@ -20,8 +20,29 @@ Go and [register a new OAuth application](https://github.com/settings/applicatio
 
 Once you've created the application, you should be given the client id and the client secret. **Make a note of these**.
 
+Additionally, if the intention is to run the GitHub repository poller (`npm run poll-repositories`), a API token should be acquired so the API calls are submitted to an acceptable rate-limit (5k req/s instead of 60 req/s for anonymous access)::
+
+  curl -s -u '<GH_USER>' https://api.github.com/authorizations \
+    -H 'Content-Type: application/json' \
+    -H 'X-GitHub-OTP: <GH 2FA CODE IF ENABLED>' \
+    -d '{"client_id": "c54ee622a3fa9622701f", \
+         "client_secret": "2a6010d06c7c90b1fcf6518d72eac0bab64cf35b", \
+	 "note": "Build.s.io poller token"}' | jq '.token' -r
+
+`curl` will ask for you GitHub password (basic-auth) and acquire a 40-char authorization token. Make note of it, it should be set in the running environment as `GITHUB_AUTH_CLIENT_TOKEN`.
```

Comment:
In theory you don't need to generate the token. BSI is a registered app, so it can use its own rate limit just by sending client_id and client_secret with in query of API request: https://developer.github.com/v3/#increasing-the-unauthenticated-rate-limit-for-oauth-applications

We use that in BSI to update snapcraft.yaml data when we recieve push from webhook.

So, talking about rate limits.
It seems that the only place where BSI calls GH on its own (via client_id + client_secret) is when we get webhook push and we update the status of snapcraft.yaml data (to make sure we have a fresh snap name in case it was updated in a commit). So thats the only action that affects BSI own rate limit (5k/h).

If poller script shares the credentials it would take from the same rate limit. That would mean that if pooler sends more than 5k req to GH in an hour BSI will not respond correctly to webhooks.

But, from my understanding, if poller uses separate token (not client_id + client_secret) pair it is likely to have a separate rate limit for that token.

So, on one hand poller script auth may be simplified to not require separate token and just use client_id and client_secret, but that would mean shared rate limit between BSI webapp and poller script.
Having separate token for poller script means more configuration (and need to request the token, authenticate to GH, etc), but it should make sure that webapp and poller have separate rate limits.